lib/optimizely/optimizely_factory.rb

```diff
@@ -19,38 +19,51 @@
 require 'optimizely'
 module Optimizely
   class OptimizelyFactory
-    def self.create_default_instance_with_sdk_key(sdk_key)
-      Optimizely::Project.new(nil, nil, nil, nil, nil, nil, sdk_key)
+    # Returns a new optimizely instance.
+    #
+    # @params sdk_key - Required String uniquely identifying the fallback datafile corresponding to project.
+    # @param fallback - Optional JSON string datafile.
+    def self.default_instance(sdk_key, fallback = nil)
+      Optimizely::Project.new(fallback, nil, nil, nil, nil, nil, sdk_key)
     end
 
-    def self.create_default_instance_with_sdk_key_and_datafile(sdk_key, datafile = nil)
-      Optimizely::Project.new(datafile, nil, nil, nil, nil, nil, sdk_key)
-    end
-
-    def self.create_default_instance_with_config_manager(config_manager)
+    # Returns a new optimizely instance.
+    #
+    # @param config_manager - Required ConfigManagerInterface Responds to get_config.
+    def self.default_instance_with_config_manager(config_manager)
       Optimizely::Project.new(nil, nil, nil, nil, nil, nil, nil, config_manager)
     end
 
-    def self.create_default_instance(
-      datafile = nil,
+    # Returns a new optimizely instance.
+    #
+    # @params sdk_key - Required String uniquely identifying the datafile corresponding to project.
+    # @param fallback - Optional JSON string datafile.
+    # @param event_dispatcher - Optional EventDispatcherInterface Provides a dispatch_event method which if given a URL and params sends a request to it.
+    # @param logger - Optional LoggerInterface Provides a log method to log messages. By default nothing would be logged.
+    # @param error_handler - Optional ErrorHandlerInterface which provides a handle_error method to handle exceptions.
+    #                 By default all exceptions will be suppressed.
+    # @param skip_json_validation - Optional Boolean param to skip JSON schema validation of the provided datafile.
+    # @param user_profile_service - Optional UserProfileServiceInterface Provides methods to store and retreive user profiles.
+    # @param notification_center - Optional Instance of NotificationCenter.
+    def self.custom_instance(
+      sdk_key,
+      fallback = nil,
       event_dispatcher = nil,
       logger = nil,
       error_handler = nil,
       skip_json_validation = false,
       user_profile_service = nil,
-      sdk_key = nil,
-      config_manager = nil,
```

Comment:
We should keep this config manager and pass it along to the constructor. We just shouldn't make it mandatory, default it to nil